db/migrate/20210211115125_migrate_shipment_fees_to_shipments.rb

```diff
@@ -0,0 +1,18 @@
+class MigrateShipmentFeesToShipments < ActiveRecord::Migration
+  def up
+    # Shipping fee adjustments currently have the order as the `adjustable` and the shipment as
+    # the `source`. Both `source` and `adjustable` will now be the shipment. The `originator` is
+    # the shipping method, and this is unchanged.
+    Spree::Adjustment.where(originator_type: 'Spree::ShippingMethod').update_all(
+      "adjustable_id = source_id, adjustable_type = 'Spree::Shipment'"
+    )
+  end
+
+  def down
+    # Just in case: reversing this migration requires setting the `adjustable` back to the order.
+    # The type is 'Spree::Order', and the order's id is still available on the `order_id` field.
+    Spree::Adjustment.where(originator_type: 'Spree::ShippingMethod').update_all(
+      "adjustable_id = order_id, adjustable_type = 'Spree::Order'"
+    )
+  end
+end
```

Comment:
So *this* migration is reversible, but a future migration (where we actually reduce the number of polymorphic associations from three to two) would *not* be reversible, right?

Reply:
I think that migration will be reversible, when we get to it... :+1: